### PR TITLE
Refactor all the things

### PR DESCRIPTION
**Pull request by kaylynn234**
There's a lot happening in this one. So here's a few bullet points:
- AbstractEventGatherer for gathering events nicely (ext.menus style)
- Move some stuff into YAML
- Cleanup of existing YAML 
- Big madlibs refactor
- Custom context to support usage of AbstractEventGatherer
- New exception for AbstractEventGatherer called StopHandler, similar in purpose to StopIteration
- More, probably

This needs so much review. Help.

### CODE REVIEW

File: crimsobot/handlers/games.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+import discord
+
+from crimsobot.data.games import CRINGO_RULES, EMOJISTORY_RULES
+from crimsobot.handlers import AbstractEventGatherer, StopHandler, must_be_event
+from crimsobot.utils import cringo, tools as c
+
+
+@must_be_event('on_reaction_add')
+class CringoJoinHandler(AbstractEventGatherer):
+    def can_join_cringo(
```
Comment (williammck) [inline crimsobot/handlers/games.py:12]: non-blocking nit: should the internal methods of these handlers should be prefixed with underscores?

File: crimsobot/bot.py [hunk 2]
```diff
@@ -189,17 +190,17 @@ async def on_message(self, message: discord.Message) -> None:
                 ])
             )
 
+        # Get context from message for command invocation and text generation
+        ctx = await self.get_context(message, cls=CrimsoContext)
+
         # process commands
-        await self.process_commands(message)
+        await self.invoke(ctx)
```
Comment (williammck) [inline crimsobot/bot.py:197]: we were inadvertently calling get_context twice? nice 😂

**Review by williammck — APPROVED** [review]
This looks great.
I think as this grows we may want to split up the Games handlers file, but I understand that it's like this now as it's all the handlers for the Games cog.
At some point I'd like to chase down how to make mypy understand all of this, but for right now I'm not too fussed by it.

**Review by williammck — APPROVED** [review]

File: crimsobot/bot.py (hunk 2 above)
Comment (kaylynn234) [reply crimsobot/bot.py:197]: yep!

File: cli.py [hunk 3]
```diff
@@ -26,7 +26,7 @@ def run():
     """This command runs the bot."""
 
     paginator = Paginator(max_size=1336)
-    bot = CrimsoBOT(help_command=PaginatedHelpCommand(paginator=paginator))
+    bot = CrimsoBOT(help_command=PaginatedHelpCommand(paginator=paginator), case_insensitive=True)
```
Comment (williammck) [inline cli.py:29]: hmmm, why not just put this in the `__init__` call in the actual `CrimsoBOT` class?

Comment (kaylynn234) [reply cli.py:29]: good point

**Review by williammck — APPROVED** [review]

**Pull request merged**